### ISSUE TITLE
[duckdb] Unable to parse expressions in Intervals

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7377]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

In dynamic intervals, parentheses are "necessary for variable amounts": https://duckdb.org/docs/stable/sql/data_types/interval

SQLFluff will fail to parse statements that include parenthesized intervals. 

### Expected Behaviour

Parenthesized intervals should be parsed

### Observed Behaviour

> L:   3 | P:  45 |  PRS | Line 3, Position 45: Found unparsable section: 'as
>                        | col2,'

### How to reproduce

```sql
SELECT
    today() + INTERVAL 1 YEAR as col1,
    today() + INTERVAL (1000* random()) day as col2,
```
`uv run sqlfluff lint --dialect duckdb test1.sql`
...
> L:   3 | P:  45 |  PRS | Line 3, Position 45: Found unparsable section: 'as
>                        | col2,'

### Dialect

duckdb

### Version

4.0.0

### Configuration

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"
sql_file_exts = ".sql,.sql .j2,.dml,.ddl"
exclude_rules = [
    "AM04",
    "ST06",
    "LT01",
    "LT12",
    "JJ01"
]

[tool.sqlfluff.indentation]
tab_space_size = 2

[tool.sqlfluff.rules.capitalisation.keywords]
capitalisation_policy = "upper"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by franloza** [comment]
I created a PR to fix this bug: https://github.com/sqlfluff/sqlfluff/pull/7388
